### PR TITLE
chore: Update configuration logic

### PR DESCRIPTION
**Pull request by markbeierl**
# Description

Changes the bessd config check from file presence based to query process state.

# Checklist:

- [X] My code follows the [style guidelines](/CONTRIBUTING.md) of this project
- [X] I have performed a self-review of my own code
N/A I have made corresponding changes to the documentation
- [x] I have added tests that validate the behaviour of the software
- [X] I validated that new and existing unit tests pass locally with my changes
N/A Any dependent changes have been merged and published in downstream modules
N/A I have bumped the version of the library

### CODE REVIEW

File: src/charm.py [hunk 1]
```diff
@@ -640,65 +649,95 @@ def _configure_bessd_workload(self) -> None:
...
+            (stdout, stderr) = self._exec_command_in_bessd_workload(
+                command=command,
+                environment=self._bessd_environment_variables,
+                timeout=30,
+            )
+
+            logger.info("Service `bessd` configuration script complete")
+            for line in stdout.splitlines():
+                logger.debug("`up4.bess`: %s", line)
+            if isinstance(stderr, str) and len(stderr) > 0:
```
Comment (ghislainbourgeois) [inline src/charm.py:672]: The return value of `_exec_command_in_bessd_workload` is a tuple of strings, so `stderr` should always be a string. This line could be simplified to this:
```suggestion
            if stderr:
```

File: src/charm.py [hunk 2]
```diff
@@ -640,65 +649,95 @@ def _configure_bessd_workload(self) -> None:
...
+        command = "/opt/bess/bessctl/bessctl show worker"
+        try:
+            (stdout, _) = self._exec_command_in_bessd_workload(
+                command=command,
+                timeout=10,
+            )
+            logger.debug("bessd configured workers: %s", stdout)
+            return True
+        except ExecError as e:
+            logger.info(f"Configuration check: {e}")
```
Comment (ghislainbourgeois) [inline src/charm.py:728]: The code style guide disallows f-strings in logging calls.
```suggestion
            logger.info("Configuration check: %s", e)
```

**Review by ghislainbourgeois — CHANGES_REQUESTED** [review]
Looks good, only a couple of small comments.

File: src/charm.py (hunk 1 above)
Comment (markbeierl) [reply src/charm.py:672]: One of the linters was complaining about it in the k8s charm, and I had to use that syntax there. I'll try it to see if the signature is different here and will pass

Comment (ghislainbourgeois) [reply src/charm.py:672]: Ok, if this suggestion does not work let me know, this is not a big concern.

**Review by ghislainbourgeois — APPROVED** [review]

**Pull request merged**